internal/cortex/chunk/encoding/bigchunk.go

```diff
@@ -278,7 +278,7 @@ func (it *bigchunkIterator) FindAtOrAfter(target model.Time) bool {
 		it.curr = it.chunks[it.i].Iterator(it.curr)
 	}
 
-	for it.curr.Next() {
+	for it.curr.Next() != chunkenc.ValNone {
```

Comment:
Also, wouldn't it be a bit more robust to check:
```suggestion
	for it.curr.Next() != chunkenc.ValFloat {
```
? :thinking: This way, we would skip over histograms.